ringbuffer/ringbuffer.go

```diff
@@ -286,3 +286,11 @@ func (rb *RingBuffer) DiscardAll() (err error) {
 	rb.desc.readPointer = rb.desc.writePointer
 	return nil
 }
+
+// PacketSize returns the packetSize held in the buffer description
+func (rb *RingBuffer) PacketSize() int64 {
+	if rb.desc == nil {
+		return -1
+	}
+	return rb.desc.packetSize
+}
```

Comment:
Well, it certainly exists for a purpose: to tell us what size the device's packets are. (Assumption: they will never change size. If that rule is incorrect, much new code will be needed!)

In response to your comment, I decided not to change the code here, but instead change `abaco.go` at the point where we _call_ this method. Instead of ignoring the possible negative value, I check it. It would admittedly be more go-ish to return `(int64, error)`, but I decided not to do that.

Reply:
I strongly feel we should do the more-goish way, to the point that I will not click the approve button if we do not. I do not want to mix error handling paradigms, and as written, it is very easy to use that function without checking the error, as you were doing. The goish way will force you to at least write down a variable for the error if you want the packet size. 

The reason I said it "only exists to cause uncaught errors" is that direct field access is another way to get the same info, and it would error on `rb.desc == nil`. So the addition of the function, over direct field access, hides errors that direct field access would reveal.

Reply:
Fine. See 909b88f. Realistically any problems of this sort would already have generated errors, in our specific case.  You're right, though, about being more explicit--who knows if future use cases might be different.